streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java

```diff
@@ -357,7 +357,9 @@ private void handleCheckPoints(Application application) throws Exception {
     FlinkCluster flinkCluster = getFlinkCluster(application);
     CheckPoints checkPoints = httpCheckpoints(application, flinkCluster);
     if (checkPoints != null) {
-      checkpointProcessor.process(application.getId(), checkPoints);
+      checkPoints
+          .getLatestChkAndSavepointWithTimeAscOrder()
```

Comment:
Why `getLatestChkAndSavepointWithTimeAscOrder ` return a list? And why need process all checkpoints and savepoints?  The  latest checkpoint and latest savepoint should be enough,  right?

Reply:
The original intention is that get chk & savepoint from`latest` with order, then records the info into db. The latest savepoint or chk will be tagged with 'latest'.

Reply:
I prefer move the logic into  `checkpointProcessor`, this class is `FlinkRESTAPIWatcher`, it shouldn't care about these logics.

Overall, don't change this file, and we still pass the `CheckPoints checkPoints` to `checkpointProcessor.process(application.getId(), checkPoints);`.